src/main/java/org/terasology/web/servlet/JsonWebApplicationException.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.web.servlet;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+public class JsonWebApplicationException extends WebApplicationException {
```

Comment:
I'm not sure I get what you mean. A `JsonWebApplicationException` is constructed by specifying a status code and a message, which is sent as the HTTP response code, which is sent in a JSON object like `{"message": "the error message"}`. They are provided by the code which builds the exception.

I can't figure out a better way but probably it's just me, if you think there is one could you please be more specific?